v0.10 migration guides (#226)
infracost/docs: docs/guides/atlantis_migration.md

import useBaseUrl from '@docusaurus/useBaseUrl';

:::tip
This is the migration guide for the upcoming v0.10 - which has not been released yet.
A [beta](https://github.com/infracost/infracost/releases/tag/v0.10.0-beta.1) is available if you'd like to try an early version.
:::

Follow this page to migrate your [Infracost Atlantis integration](https://github.com/infracost/infracost-atlantis).

If you encounter any issues while migrating, please join our [community Slack channel](https://www.infracost.io/community-chat), we'll help you very quickly 😄

## What's new?

With the v0.10 release, we'll support two ways to run Infracost with Terraform via `--path`:

> 1. **Parsing HCL code**: this is the default option as it has [4 key benefits](/docs/guides/v0.10_migration/#1-faster-cli). This page describes how you can migrate to this option.

Reviewer: ```suggestion
1. **Parsing HCL code**: this is the default CLI option as it has [4 key benefits](/docs/guides/v0.10_migration/#1-faster-cli).
```

    ```shell
    # Terraform variables can be set using --terraform-var-file or --terraform-var
    infracost breakdown --path /code
    ```

> 2. **Parsing plan JSON file**: this will continue to work as before and is recommended for deployment systems such as Atlantis which are already generating a plan JSON.

Reviewer: ```suggestion
2. **Parsing plan JSON file**: this will continue to work as before and is recommended for deployment systems such as Atlantis which are already generating a plan JSON.  This page describes how you can migrate to this option.

## Infracost Atlantis migration guide

1. If you are using the `infracost-atlantis` Docker image update the tag to use v0.10 of Infracost. For examples if you were using `infracost/infracost-atlantis:atlantis0.19-infracost0.9` Docker image you should update to `infracost/infracost-atlantis:atlantis0.19-infracost0.10`.

    If you already use a custom Docker image for Atlantis, change the line that's installing Infracost to download from `https://infracost.io/downloads/v0.10/infracost-linux-amd64.tar.gz`.

> 2. If you are calling Infracost with `--path=$PLANFILE` you should change this to `--plan=$SHOWFILE` to pass the plan JSON file instead of the plan binary file, since this will be supported long term. This might require you to add a `show` step after the `plan` step of the workflow, e.g:

Reviewer: ```suggestion
2. If you are calling Infracost with `--path=$PLANFILE` you should change this to `--plan=$SHOWFILE` to pass the plan JSON file instead of the plan binary file, since Infracost will be supporting that long term. This might require you to add a `show` step after the `plan` step of the workflow, e.g:
```
